test/spmd/tensor/parallel/test_parallelize_api.py

```diff
@@ -126,6 +134,161 @@ def __init__(self) -> None:
                 torch.nn.Linear(10, 5), device_mesh, PairwiseParallel()
             )
 
+    @with_comms
+    def test_linear_row_wise_parallel(self):
+        # test RowwiseParallel
+        inp_size = [8, 16]
+        torch.manual_seed(self.rank)
+        inp = torch.rand(*inp_size, device=self.device_type)
+        rowwise = RowwiseParallel()
+
+        torch.manual_seed(5)
+        model = torch.nn.Linear(16, 10, device=self.device_type)
+        torch.manual_seed(5)
+        model_tp = torch.nn.Linear(16, 10, device=self.device_type)
+
+        # parallelize model_tp
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        _parallelize_linear(model_tp, device_mesh, rowwise)
+
+        self.assertEqual(
+            distribute_tensor(model.weight, device_mesh, [Shard(1)]).to_local(),
+            model_tp.weight.to_local(),
+        )
+        self.assertEqual(
+            distribute_tensor(
+                model.bias, device_mesh, [Replicate()]
+            ).to_local(),
+            model_tp.bias.to_local(),
+        )
+
+        LR = 0.25
+        optim = torch.optim.SGD(model.parameters(), lr=LR)
+        optim_tp = torch.optim.SGD(model_tp.parameters(), lr=LR)
+
+        local_inp = make_input_replicate_1d(
+            DTensor.from_local(inp, device_mesh, [Shard(0)])
+        ).to_local()
+        output = model(local_inp)
+        output_tp = model_tp(inp)
+        self.assertEqual(output, output_tp.to_local())
+
+        output.sum().backward()
+        output_tp.sum().backward()
+
+        replicate = [Replicate()]
+        # Ensure gradients are same.
+        self.assertEqual(
+            model.weight.grad,
+            model_tp.weight.grad.redistribute(
+                device_mesh=device_mesh, placements=replicate
+            ).to_local(),
+        )
+        self.assertEqual(
+            model.bias.grad,
+            model_tp.bias.grad.redistribute(
+                device_mesh=device_mesh, placements=replicate
+            ).to_local(),
+        )
+
+        optim.step()
+        optim_tp.step()
+
+        # Ensure params are same.
+        self.assertEqual(
+            model.weight,
+            model_tp.weight.redistribute(
+                device_mesh=device_mesh, placements=replicate
+            ).to_local(),
+        )
+        self.assertEqual(
+            model.bias,
+            model_tp.bias.redistribute(
+                device_mesh=device_mesh, placements=replicate
+            ).to_local(),
+        )
+
+        inp = torch.rand(*inp_size, device=self.device_type)
+        local_inp = make_input_replicate_1d(
+            DTensor.from_local(inp, device_mesh, [Shard(0)])
+        ).to_local()
+        output = model(local_inp)
+        output_tp = model_tp(inp)
+        self.assertEqual(output, output_tp.to_local())
+
+    @with_comms
+    def test_linear_col_wise_parallel(self):
```

Comment:
Can we consolidate the two test a little bit, the code is way too long and there are multiple lines of duplication. Thanks! Feel free to refactor the whole test file to make it cleaner.

Reply:
done. TODO: see if existing test code in `test_tp_example.py` can be replaced as well.

Reply:
Please add TODO in the code if you have not done it, thanks!